lib/helpers/async-with-concurrency.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-loop-func */
+
+'use strict';
+
+module.exports = class AsyncWithConcurrency {
+
+	constructor(workerFn, maxConcurrency) {
+		this.workerFn = workerFn;
+		this.maxConcurrency = maxConcurrency;
+		this.activeTasks = 0;
+	}
+
+	async run(queue) {
```

Comment:
@merq-rodriguez con este metodo tengo dudas, esta medio raro como lo hiciste, con el this.activeTasks y con el promise.race, por que supone que la primera vez a while que entra al while y tenes 2 tasks esta mete 2 promises y luego entra otra vez la primera vez y en la segunda otra vez se hace otra por que this.activeTask es 1 todavia?
No podriamos no depender de activeTask y usar un promise.allSettled?